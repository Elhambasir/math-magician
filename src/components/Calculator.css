.container {
  width: 300px;
  background-color: lightblue;
  display: flex;
  flex-direction: column;
}

.input-data {
  display: flex;
  flex-direction: column;
}

.input-data input {
  border: none;
  outline: none;
  direction: rtl;
  color: white;
  height: 30px;
  background-color: rgba(62, 58, 66, 0.934);
  padding-right: 10px;
}

.buttons {
  display: flex;
}

.buttons .left-side {
  flex: 2;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
}

.buttons .left-side > * {
  padding: 10px 25px;
  border: 1px solid lightgray;
  font-size: 8px;
  color: black;
  font-weight: bold;
  cursor: pointer;
}

.buttons .left-side .zero {
  grid-column: 1 / span 2;
}

.buttons .right-side {
  flex: 1;
  display: grid;
  flex-direction: column;
}

.buttons .right-side > * {
  padding: 10px 25px;
  background-color: orange;
  border: 1px solid lightgray;
  font-size: 8px;
  color: black;
  font-weight: bold;
  cursor: pointer;
}
