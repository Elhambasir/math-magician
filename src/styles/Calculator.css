main {
  display: flex;
  justify-content: center;
}

.container {
  width: 500px;
  display: flex;
  flex-direction: column;
  height: 500px;
}

.input-data {
  display: flex;
  flex-direction: column;
}

.input-data input {
  border: none;
  outline: none;
  direction: rtl;
  color: white;
  height: 50px;
  background-color: rgba(62, 58, 66, 0.934);
  padding-right: 10px;
}

.buttons {
  width: 500px;
  height: 300px;
  display: flex;
}

.buttons .left-side {
  flex: 3;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
}

.buttons .left-side > * {
  padding: 10px 25px;
  border: 1px solid lightgray;
  font-size: 14px;
  color: gray;
  font-weight: bold;
  cursor: pointer;
}

.buttons .left-side .zero {
  grid-column: 1 / span 2;
}

.buttons .right-side {
  flex: 1;
  display: grid;
  flex-direction: column;
}

.buttons .right-side > * {
  padding: 10px 25px;
  background-color: orange;
  border: 1px solid lightgray;
  font-size: 14px;
  color: gray;
  font-weight: bold;
  cursor: pointer;
}

.buttons .left-side button:hover,
.buttons .right-side button:hover {
  box-shadow: 0 10px 15px -3px rgba(0, 0, 0, 0.1);
  border: 2px solid gray;
}
