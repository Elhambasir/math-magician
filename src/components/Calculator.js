/* eslint-disable react/prop-types */
import React, { useState } from 'react';
import './Calculator.css';
import calculate from './logic/calculate';
import Quote from './quote';

const MyCulc = () => {
  const [state, updateState] = useState({
    total: null,
    next: null,
    operation: null,
  });
  const calc = (e) => {
    const newState = calculate(state, e.target.textContent);
    updateState(newState);
  };
  return (
    <main>
      <div className="container">
        <div className="input-data">
          <InputData data={state} />
        </div>
        <div className="buttons">
          <div className="left-side">
            <button type="button" className="ac" onClick={calc}>AC</button>
            <button type="button" onClick={calc} className="plusOrMinus">+/-</button>
            <button type="button" onClick={calc} className="modelus">%</button>
            <button type="button" onClick={calc} className="seven">7</button>
            <button type="button" onClick={calc} className="eight">8</button>
            <button type="button" onClick={calc} className="nine">9</button>
            <button type="button" onClick={calc} className="four">4</button>
            <button type="button" onClick={calc} className="five">5</button>
            <button type="button" onClick={calc} className="six">6</button>
            <button type="button" onClick={calc} className="1">1</button>
            <button type="button" onClick={calc} className="two">2</button>
            <button type="button" onClick={calc} className="three">3</button>
            <button type="button" onClick={calc} className="zero">0</button>
            <button type="button" onClick={calc} className="dot">.</button>
          </div>
          <div className="right-side">
            <button type="button" onClick={calc} className="devid">/</button>
            <button type="button" onClick={calc} className="times">X</button>
            <button type="button" onClick={calc} className="minus">-</button>
            <button type="button" onClick={calc} className="plus">+</button>
            <button type="button" onClick={calc} className="equal">=</button>
          </div>
        </div>
      </div>
      <Quote />
    </main>
  );

  function InputData(props) {
    const state = props;
    return (
      <div className="input-data">
        <input type="text" name="input-data" value={state.data.next || state.data.total || 0} readOnly disabled />
      </div>
    );
  }
};
export default MyCulc;
