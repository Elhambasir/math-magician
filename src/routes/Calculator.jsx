import MyCulc from '../components/MyCulc';
import styles from '../styles/Calculator.module.css';

const Calculator = () => (
  <>
    <div className={styles.wrapper}>
      <div className={styles.content}>
        <h2> Lets do some math!</h2>
      </div>
      <div className={styles.calculator}>
        <MyCulc />
      </div>
    </div>
  </>
);
export default Calculator;
