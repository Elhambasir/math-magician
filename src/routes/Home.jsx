import styles from '../styles/Home.module.css';

const Home = () => (
  <>
    <div className={styles.wrapper}>
      <div className={styles.title}>
        <h2>Home</h2>
      </div>
      <div className={styles.content}>
        <div>
          <p>
            Lorem ipsum dolor sit, amet consectetur
            adipisicing elit. Dolores quo labore rerum
            sunt debitis vitae odio mollitia neque,
            accusamus rem optio saepe corporis ipsa
            necessitatibus eaque,
            culpa ratione quidem! Repellendus.
            Lorem ipsum dolor sit, amet consectetur
            adipisicing elit. Dolores quo labore rerum
            sunt debitis vitae odio mollitia neque,
            accusamus rem optio saepe corporis ipsa
            necessitatibus eaque,
            culpa ratione quidem! Repellendus.
          </p>
        </div>
        <div>
          <p>
            Lorem ipsum dolor sit, amet consectetur
            adipisicing elit. Dolores quo labore rerum
            sunt debitis vitae odio mollitia neque,
            accusamus rem optio saepe corporis ipsa
            necessitatibus eaque,
            culpa ratione quidem! Repellendus.
            Lorem ipsum dolor sit, amet consectetur
            adipisicing elit. Dolores quo labore rerum
            sunt debitis vitae odio mollitia neque,
            accusamus rem optio saepe corporis ipsa
            necessitatibus eaque,
            culpa ratione quidem! Repellendus.
          </p>
        </div>
      </div>
    </div>
  </>
);
export default Home;
