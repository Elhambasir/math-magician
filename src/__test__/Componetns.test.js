import React from 'react';
import { render } from '@testing-library/react';
import Home from '../routes/Home';
import Quote from '../routes/quote';
import Calculator from '../routes/Calculator';

describe('Component render testing', () => {
  it('Home page renders correctly', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Quote page renders correctly', () => {
    const { asFragment } = render(<Quote />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Calculator renders correctly', async () => {
    const { asFragment } = render(<Calculator />);
    expect(asFragment()).toMatchSnapshot();
  });
});
