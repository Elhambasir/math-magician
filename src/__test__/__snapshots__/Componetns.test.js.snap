// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component render testing Calculator renders correctly 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      class="content"
    >
      <h2>
         Lets do some math!
      </h2>
    </div>
    <div
      class="calculator"
    >
      <main>
        <div
          class="container"
        >
          <div
            class="input-data"
          >
            <div
              class="input-data"
            >
              <input
                data-testid="result"
                disabled=""
                id="input"
                name="input-data"
                readonly=""
                type="text"
                value="0"
              />
            </div>
          </div>
          <div
            class="buttons"
          >
            <div
              class="left-side"
            >
              <button
                class="ac"
                type="button"
              >
                AC
              </button>
              <button
                class="plusOrMinus"
                type="button"
              >
                +/-
              </button>
              <button
                class="modelus"
                type="button"
              >
                %
              </button>
              <button
                class="seven"
                type="button"
              >
                7
              </button>
              <button
                class="eight"
                type="button"
              >
                8
              </button>
              <button
                class="nine"
                type="button"
              >
                9
              </button>
              <button
                class="four"
                type="button"
              >
                4
              </button>
              <button
                class="five"
                type="button"
              >
                5
              </button>
              <button
                class="six"
                type="button"
              >
                6
              </button>
              <button
                class="1"
                type="button"
              >
                1
              </button>
              <button
                class="two"
                type="button"
              >
                2
              </button>
              <button
                class="three"
                type="button"
              >
                3
              </button>
              <button
                class="zero"
                type="button"
              >
                0
              </button>
              <button
                class="dot"
                type="button"
              >
                .
              </button>
            </div>
            <div
              class="right-side"
            >
              <button
                class="devid"
                type="button"
              >
                /
              </button>
              <button
                class="times"
                type="button"
              >
                X
              </button>
              <button
                class="minus"
                type="button"
              >
                -
              </button>
              <button
                class="plus"
                type="button"
              >
                +
              </button>
              <button
                class="equal"
                type="button"
              >
                =
              </button>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Component render testing Home page renders correctly 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      class="title"
    >
      <h2>
        Home
      </h2>
    </div>
    <div
      class="content"
    >
      <div>
        <p>
          Lorem ipsum dolor sit, amet consectetur adipisicing elit. Dolores quo labore rerum sunt debitis vitae odio mollitia neque, accusamus rem optio saepe corporis ipsa necessitatibus eaque, culpa ratione quidem! Repellendus. Lorem ipsum dolor sit, amet consectetur adipisicing elit. Dolores quo labore rerum sunt debitis vitae odio mollitia neque, accusamus rem optio saepe corporis ipsa necessitatibus eaque, culpa ratione quidem! Repellendus.
        </p>
      </div>
      <div>
        <p>
          Lorem ipsum dolor sit, amet consectetur adipisicing elit. Dolores quo labore rerum sunt debitis vitae odio mollitia neque, accusamus rem optio saepe corporis ipsa necessitatibus eaque, culpa ratione quidem! Repellendus. Lorem ipsum dolor sit, amet consectetur adipisicing elit. Dolores quo labore rerum sunt debitis vitae odio mollitia neque, accusamus rem optio saepe corporis ipsa necessitatibus eaque, culpa ratione quidem! Repellendus.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Component render testing Quote page renders correctly 1`] = `
<DocumentFragment>
  <div>
    Loading...
  </div>
</DocumentFragment>
`;
